### PR TITLE
Getting Started with the Input-Output Streams 

### LINKED ISSUE #3621: Understanding PHP 8+ I/O Streams
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
Understanding PHP I/O Streams

### Introduction paragraph (2-3 paragraphs):
One of the benefits of PHP being a server side scripting environment is that it gives the web developer easy access to the filesystem of the server on which the web server is running. This gives us the ability to create, open, delete, read and write to files. We can also traverse the directory hierarchy of the server systems to do things like get directory listings and create new sub-directories.

Existing files are opened, and new files created using the PHP fopen() function. The fopen() function accepts two arguments and returns a file handle which is subsequently used for all future read and write interactions with that file. The first argument is the name (including the path) of the file to open. This path is relative to the server filesystem root, not your web server root.  

### Key takeaways: 
1.Writing to a File using PHP
2.Reading From a File using PHP
3.Checking Whether a File Exists
4.Moving, Copying and Deleting Files with PHP
5.Accessing File Attributes
### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by TheresaAtieno**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes
This closes #3621

### CODE REVIEW

**Comment by jumamiller** [comment]
Hi @TheresaAtieno, welcome to @Section.
I'm looking forward to reviewing this article as soon as possible!

Thank you!

**Comment by WanjaMIKE** [comment]
Plagiarism check done.

**Comment by TheresaAtieno** [comment]
> Hi @TheresaAtieno, welcome to @section. I'm looking forward to reviewing this article as soon as possible!
> 
> Thank you!

@jumamiller how do I add my author details? Check your slack I sent you a message there too!

**Review by jumamiller — APPROVED** [review]
Great add, kindly add your author details for us to proceed to final review!

**Comment by TheresaAtieno** [comment]
@jumamiller done!

**Review by jumamiller — APPROVED** [review]

**Comment by jumamiller** [comment]
Just what I needed for the school project! Thank you @TheresaAtieno for this informative article on streams in PHP.

Once again, welcome to the Section community, looking forward for more articles on PHP.

**Comment by lalith1403** [comment]
Jumping on this one 🚀

File: content/articles/php-io-streams/index.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+When you execute the above code, you notice that it doesn't give you an error, as was the case when we used the `file_get_contents()` function. This is the power of streams.
+
+With streams, we are able to read any file size without running into memory management problems which maybe very frustrating to a developer.
+
+### Retrieving file contents
+In the previous section, we have seen how we can use the power of streams in PHP to handle large files, in our case, using the `fopen()` function. 
+
+Now, let's understand how we can retrieve the file's contents using the `fgets()` function.
+```php
+...
```
Comment (lalith1403) [inline content/articles/php-io-streams/index.md:121]: check this code section. do we need the extra ... before and after the code snippets?

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Address the comment :)

**Comment by lalith1403** [comment]
ready to merge once question is clarified.

File: content/articles/php-io-streams/index.md (hunk 1 above)
Comment (TheresaAtieno) [reply content/articles/php-io-streams/index.md:121]: yes, we need them, to show that some imports are needed for the code to work!

**Comment by TheresaAtieno** [comment]
> ready to merge once the question is clarified.

Hello sir, the comment addressed! Thank you!

**Pull request merged**